x-pack/plugin/sql/qa/server/src/main/resources/datetime-custom-tz.csv-spec

```diff
@@ -0,0 +1,72 @@
+
+// Tests using connection specific timezones (according to the `-TZ[...]` suffixes).
+//
+// Most of the tests cast timestamps to strings on the server side because of the way JDBC handles datetimes.
+// The issue is also described in https://github.com/elastic/elasticsearch/issues/40779.
+//
+// Note: Etc/GMT-1 is equivalent to offset +01:00 but JDBC requires timezone ids.
+
+castStringWithoutTZToDateTime-TZ[Etc/GMT-1]
+SELECT CAST(CAST('2020-06-01T09:20:30' AS DATETIME) AS STRING) t;
+
+           t:s
+-----------------------------
+2020-06-01T09:20:30.000+01:00
+;
+
+castIntToDateTime-TZ[Etc/GMT-1]
+SELECT CAST(CAST(0 AS DATETIME) AS STRING) t;
+
+           t:s
+-----------------------------
+1970-01-01T00:00:00.000Z
+;
+
+castStringWithTZToDateTime-TZ[Etc/GMT-1]
+SELECT ('2020-06-01T09:20:30Z'::DATETIME)::STRING a,
+       ('2020-06-01T09:20:30+01:00'::DATETIME)::STRING b,
+       ('2020-06-01T09:20:30-10:30'::DATETIME)::STRING c;
+
+           a:s          |           b:s               |           c:s
+------------------------+-----------------------------+-----------------------------
+2020-06-01T09:20:30.000Z|2020-06-01T09:20:30.000+01:00|2020-06-01T09:20:30.000-10:30
+;
+
+castStringWithoutTZToDatetimeInScript-TZ[Etc/GMT-1]
+SELECT hire_date::string,
```

Comment:
Supernit: the rest of the tests use capitalised `::STRING`, but I was actually wondering if the conversion is [always] necessary?

Reply:
👍 for the capitalization.

But yes, as soon as you get timestamps in another timezone as UTC, the JDBC issue mentioned in https://github.com/elastic/elasticsearch/issues/40779 surfaces.